### PR TITLE
Fix incorrect doc in tf.nn.convolution 

### PR DESCRIPTION
**Pull request by silent567**
The ranks of input and filter should be N+2 instead of N.
Since the function computes N-D convolution, the input's rank should be N+2 with additional batch and input_channel dimensions, and the filter's rank should be N+2 with additional input_channel and output_channel dimensions.
The function actually requires the length of dilation_rate/stides to be rank(input)-2. Otherwise, it will raise ValueError.

### CODE REVIEW

**Comment by drpngx** [comment]
@jbms could you take a look?

**Comment by tensorflowbutler** [comment]
Nagging Reviewer : You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 14 days with no activity and the `awaiting review` label has been applied.

**Review by jbms — APPROVED** [review]
Thanks for catching this mistake!

**Comment by tensorflowbutler** [comment]
Nagging Assignee @drpngx: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @drpngx: It has been 44 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @drpngx: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.
